airbyte-integrations/connectors/source-github/unit_tests/test_stream.py

```diff
@@ -659,3 +660,86 @@ def get_records(cursor_field):
+def test_stream_reviews_incremental_read():
+
+    url_pulls = "https://api.github.com/repos/organization/repository/pulls"
+
+    repository_args_with_start_date = {
+        "start_date": "2000-01-01T00:00:00Z",
+        "page_size_for_large_streams": 30,
+        "repositories": ["organization/repository"],
+    }
+    stream = Reviews(parent=PullRequests(**repository_args_with_start_date), **repository_args_with_start_date)
+
+    responses.add(
+        "GET",
+        url_pulls,
+        json=[
+            {"updated_at": "2022-01-01T00:00:00Z", "number": 1},
+            {"updated_at": "2022-01-02T00:00:00Z", "number": 2},
+        ],
+    )
+
+    responses.add(
+        "GET",
+        "https://api.github.com/repos/organization/repository/pulls/1/reviews",
+        json=[{"id": 1000, "body": "commit1"}, {"id": 1001, "body": "commit1"}],
+    )
+
+    responses.add(
+        "GET",
+        "https://api.github.com/repos/organization/repository/pulls/2/reviews",
+        json=[{"id": 1002, "body": "commit1"}],
+    )
+
+    stream_state = {}
+    records = read_incremental(stream, stream_state)
+
+    assert records == [
+        {"body": "commit1", "id": 1000, "parent_updated_at": "2022-01-01T00:00:00Z", "repository": "organization/repository"},
+        {"body": "commit1", "id": 1001, "parent_updated_at": "2022-01-01T00:00:00Z", "repository": "organization/repository"},
+        {"body": "commit1", "id": 1002, "parent_updated_at": "2022-01-02T00:00:00Z", "repository": "organization/repository"},
+    ]
+
+    assert stream_state == {"organization/repository": {"parent_updated_at": "2022-01-02T00:00:00Z"}}
+
+    responses.add(
+        "GET",
+        url_pulls,
+        json=[
+            {"updated_at": "2022-01-03T00:00:00Z", "number": 1},
+            {"updated_at": "2022-01-02T00:00:00Z", "number": 2},
+            {"updated_at": "2022-01-04T00:00:00Z", "number": 3},
+        ],
+    )
+
+    responses.add(
+        "GET",
+        "https://api.github.com/repos/organization/repository/pulls/1/reviews",
+        json=[{"id": 1000, "body": "commit1"}, {"id": 1001, "body": "commit2"}],
+    )
+
+    responses.add(
+        "GET",
+        "https://api.github.com/repos/organization/repository/pulls/3/reviews",
+        json=[{"id": 1003, "body": "commit1"}],
+    )
+
+    records = read_incremental(stream, stream_state)
+
+    assert records == [
+        {"body": "commit1", "id": 1000, "parent_updated_at": "2022-01-03T00:00:00Z", "repository": "organization/repository"},
+        {"body": "commit2", "id": 1001, "parent_updated_at": "2022-01-03T00:00:00Z", "repository": "organization/repository"},
+        {"body": "commit1", "id": 1003, "parent_updated_at": "2022-01-04T00:00:00Z", "repository": "organization/repository"},
+    ]
+
+    assert stream_state == {"organization/repository": {"parent_updated_at": "2022-01-04T00:00:00Z"}}
+
+    assert len(responses.calls) == 6
+    assert urlbase(responses.calls[0].request.url) == url_pulls
```

Comment:
Could you please add some short description of what this test does / expects as output? If we just need to be sure that incremental read works as expected, the first 3 asserts completely enough, WDYT?

Reply:
short description added